c2cgeoportal/views/mapserverproxy.py

```diff
@@ -158,10 +158,17 @@ def proxy(self):
 
         role_id = None if self.user is None else \
             self.user.parent_role.id if self.external else self.user.role.id
+
+        headers = self._get_headers()
+        # Add headers for Geoserver
+        if self.settings["geoserver"] and self.user is not None:
+            headers["sec-username"] = self.user.username
+            headers["sec-roles"] = self.user.role.name
```

Comment:
There is no need to take `parent_role` into account here (as done for `role_id` above)?